src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -154,6 +158,10 @@ public static class TextLoaderSaverCatalog
             bool allowSparse = TextLoader.Defaults.AllowSparse)
         {
             Contracts.CheckNonEmpty(path, nameof(path));
+            if (!File.Exists(path))
+            {
+                throw Contracts.ExceptDecode("File does not exist at path: {0}", path);
```

Comment:
>ExceptDecode [](start = 32, length = 12)

`ExceptDecode` is used for errors in deserialization of models. I think `ExceptParam` would be more natural in this case.